firebase/test/validator.ts

```diff
@@ -0,0 +1,259 @@
+import Ajv2020 from "ajv/dist/2020";
```

Comment:
I think this implementation is looking good, although I have to add my concerns with using the term "validation/validate". I think what we should focus here is to be super specific about testing stuff. So, something like `entity-tester.ts`, as it implies that it's testing an entity and all of its properties.

As we use `Models` for (`domain/models`) objects and `Schema` (`data/schemas`) for raw data objects, I think it will fit nicely here, as we'd have `EntityTester` and the respective specializations named `SchemaTester` and `ModelTester`.

Reply:
Agreed